### PR TITLE
expose onReconnect

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

allow a onReconnect function of wsOptions

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/storybook/stories/inbox/full-page-inbox.stories.tsx [hunk 1]
```diff
@@ -36,9 +36,9 @@ export const Hooks = () => {
   useEffect(() => {
     const intervalId = setInterval(() => {
       setAuth(
-        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZSI6InVzZXJfaWQ6NzBmNmE0ZjQtMjkwNy00NTE4LWI4ZjMtYjljZmFiMjI0NzY0IGluYm94OnJlYWQ6bWVzc2FnZXMiLCJ0ZW5hbnRfc2NvcGUiOiJwdWJsaXNoZWQvcHJvZHVjdGlvbiIsInRlbmFudF9pZCI6Ijc2ODI1MWNmLTNlYjgtNDI2YS05MmViLWZhYTBlNzY3ODc2OCIsImlhdCI6MTY3MjI1NzY1OSwianRpIjoiYmJlMDMyMmMtZWY4Mi00M2FkLWI3NGMtOGZlYWNiNTczYTY0In0.Xs_yd8IhdNORK8LyleS10FDLQbb4sXkCtGHPq7tUGa4"
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZSI6InVzZXJfaWQ6NzBmNmE0ZjQtMjkwNy00NTE4LWI4ZjMtYjljZmFiMjI0NzY0IGluYm94OnJlYWQ6bWVzc2FnZXMiLCJ0ZW5hbnRfc2NvcGUiOiJwdWJsaXNoZWQvcHJvZHVjdGlvbiIsInRlbmFudF9pZCI6Ijc2ODI1MWNmLTNlYjgtNDI2YS05MmViLWZhYTBlNzY3ODc2OCIsImlhdCI6MTY3MzY1NjQwNywianRpIjoiN2IxNTZhMTEtOGMwYS00MTA0LWI3OWEtYmVmMTNhM2MzMzNlIn0.cf46AojLX2fVDZ08B9763HiQI-Rzxr-Douwd4PC52BY"
       );
-    }, 300000 * 3); // 15 minutes
+    }, 180000); // 15 minutes
```
Comment (lednax) [inline packages/storybook/stories/inbox/full-page-inbox.stories.tsx:41]: Is the comment still accurate?

**Review by lednax — APPROVED** [review]
Lgtm, question

File: packages/react-hooks/src/inbox/reducer.ts [hunk 2]
```diff
@@ -324,7 +324,10 @@ export default (state: IInbox = initialState, action?: InboxAction): IInbox => {
     }
 
     case INBOX_MARK_MESSAGE_READ: {
-      const unreadMessageCount = Math.max((state.unreadMessageCount ?? 0) - 1);
+      const unreadMessageCount = Math.max(
+        0,
+        (state.unreadMessageCount ?? 0) - 1
```
Comment (suhasdeshpande) [inline packages/react-hooks/src/inbox/reducer.ts:329]: Is there a likelihood of state being undefined? Or initial state prevents this?

File: packages/storybook/stories/inbox/full-page-inbox.stories.tsx (hunk 1 above)
Comment (rileylnapier) [reply packages/storybook/stories/inbox/full-page-inbox.stories.tsx:41]: its just in my dev env i was using to test. ill remove the comment

File: packages/react-hooks/src/inbox/reducer.ts (hunk 2 above)
Comment (rileylnapier) [reply packages/react-hooks/src/inbox/reducer.ts:329]: no, state will always exist

**Review by suhasdeshpande — APPROVED** [review]

**Pull request merged**